constellations/gemini-user/src/achievements/achievements.js

```diff
@@ -17,11 +40,40 @@ export class AchievementPage extends React.Component {
         });
     }
 
+    // {"id":1,"publishedAt":null,"year":2020,"message":"message"}
     render() {
+        const achievements = this.state.achieveList;
+
+        let sorted = {};
+        achievements.forEach((achievement) => {
+            if (!(achievement.year in sorted)) {
+                sorted[achievement.year] = [];
+            }
+            sorted[achievement.year].push(achievement);
+        });
```

Comment:
I think you do this easily using `_.groupBy` from lodash: https://lodash.com/docs/4.17.15#groupBy